articles/rock-paper-scissor-online/index.md

```diff
@@ -0,0 +1,248 @@
+        socket.emit("player1Joined",{p2name:players[data.roomID],p1name:data.name});
+    })
+```
+First the above code adds player2 to the room, next it notifies player2 with player1's info and player1 with player2's info.
+
+```
+//Player 2 Joined
+socket.on("player2Joined",(data)=>{
+    transition(data)  ;
+  })
+ 
+//Player 1 Joined
+socket.on("player1Joined",(data)=>{
+    transition(data)  ;
+})
+
+const transition=(data)=>{
+    $(".newRoom").hide();
+    $(".joinRoom").hide();
+    $(".leaderboard").show();
+    $(".controls").show();
+    $(".player1 .name").html(data.p1name);
+    $(".player2 .name").html(data.p2name);
+    $("#message").html(data.p2name+" is here!").show();
+}
+```
+ The above code simply calls the `transition()` function for both players. This `transition()` function takes care of all the UI changes to enter the game. 
+
+### Players select Choice
+
+Next we add logic for the client to be able to select a choice and emit events in the `public/game.js` file.
+
+```
+//Select Choice
+$(".controls button").click(function (){
+    const choice=$(this).html().trim();
+    const choiceEvent=firstPlayer?"choice1":"choice2";
+    socket.emit(choiceEvent,{
+        choice: choice,
+        roomID:roomID
+    });
+})
+```
+The above code gets the choice picked by the user and emits the choice Event.  The `firstPlayer` variable is used to distinguish between the kind of event the player has to emit. Player1 emits `choice1` , Player2 emits `choice2` . 
+### Listen to Player's Choice
+We add server-side side logic at `app.js` to listen to Player1's choice.
+```
+//Listener to Player 1's Choice
+    socket.on("choice1", (data)=> {
+        choice1 = data.choice;
+        console.log(choice1, choice2);
+        if (choice2 != "") {
+            result(data.roomID);
+        }
+    });
+```
+
+The above code gets player1's choice and does nothing if player2 hasn't picked their choice yet.
+
+```
+//Listener to Player 2's Choice
+    socket.on("choice2", (data)=> {
+        choice2 = data.choice;
+        console.log(choice1, choice2);
+        if (choice1 != "") {
+            result(data.roomID);
+        }
+    });
+```
+Next we write logic to listen to player2's choice in `app.js`. Once, both of player have picked their choice, the server enters into the `if block` and invokes the `result()` function.
+
+
+
+### Declare Winner
+
+In `app.js`,
+```
+//Function to be executed after getting both choices
+const result=(roomID)=> {
+    var winner = getWinner(choice1, choice2);
+    io.sockets.to(roomID).emit("result", {
+        winner: winner
+    });
+    choice1 = "";
+    choice2 = "";
+}
+```
+The result function takes the roomID as an argument. First, the `getWinner()` function calculates the result based on the player's choices .   Next, it emits the result to all the clients in the room, i.e both player1 and player2. Finally, it resets the player's choices.
```

Comment:
Extra spaces should be deleted before Next

Reply:
removed all extra spaces